### PR TITLE
Allow RichTextEditor to forwardRef

### PR DESCRIPTION
**Pull request by kendallgassner**
Issue: https://github.com/sstur/react-rte/issues/403

I want to use React-hook-form with react-rte but need to be able to forwardRef

### CODE REVIEW

**Comment by keukenkastjeXYZ** [comment]
https://github.com/sstur/react-rte/issues/403#issuecomment-910314605 maybe this can help you out, good luck!

**Review by rahul-dealroom — APPROVED** [review]

File: src/RichTextEditor.js [hunk 1]
```diff
@@ -392,3 +393,7 @@ export {
   Button,
   Dropdown,
 };
+
+export default RichTextEditor = React.forwardRef((props, ref) => (
+  <RichTextEditorBase forwardedRef={ref} {...props} />
+));
```
Comment (sstur) [inline src/RichTextEditor.js:399]: This is definitely a breaking change; anyone whose code was previously passing `forwardedRef` will stop working if we land this.

Comment (sstur) [reply src/RichTextEditor.js:399]: Also, I'm confused about the `RichTextEditor =` part here, that seems like a syntax error.

**Comment by sstur** [comment]
I'm closing this for now because:
1. Using `createRef` and `forwardRef` requires [React v16.3.0 or newer](https://reactjs.org/blog/2018/03/29/react-v-16-3.html) which will require updating the peer dependencies, which is a breaking change that I'm not ready to make at this moment (see below)
2. In order to get the tests passing, this will also require bumping Flow to [v0.89.0 or newer](https://github.com/facebook/flow/releases/tag/v0.89.0) to support `createRef` and `forwardRef`. Updating Flow surfaces a ton of new type errors and does not seem to be compatible with the version of draft-js this project is using

I'm not opposed to doing <span>#<span>1 above and bumping our major version since it's breaking, but I am opposed to <span>#</span>2 since I want to migrate away from Flow and to TypeScript instead (which will probably be less work than updating the Flow version). After we migrate to TypeScript, then I'll bump React and then we can tackle this forwardRef issue. I don't have an ETA on any of that, but I'm not planning to do a bunch of work updating Flow to get this relatively minor forwardRef feature working.

**Pull request closed without merging**